app/models/report.rb

```diff
@@ -4,33 +4,43 @@ class Report < ApplicationRecord
   validates :user_id, presence: true
   validates :item_id, presence: true
   validates :item_type, presence: true, inclusion: {
-    in: %w(Post Comment), message: "Type should match `Post` or `Comment`!"}
+    in: %w[Post Comment], message: "Type should match `Post` or `Comment`!"
+  }
   validates :text, presence: true
 
-  validate :entry_does_not_exist, :on => :create
-  validate :post_or_comment_does_exist, :on => :create
+  validate :entry_does_not_exist, on: :create
+  validate :post_or_comment_does_exist, on: :create
 
   belongs_to :user
   belongs_to :post, optional: true
   belongs_to :comment, optional: true
   belongs_to :item, polymorphic: true
 
-  after_commit :send_report_notification, :on => :create
+  STATUS_DELETED = "deleted"
+  STATUS_NO_ACTION = "no_action"
+
+  after_commit :send_report_notification, on: :create
+
+  scope :join_originator, -> {
+    joins("LEFT JOIN people ON reported_author_id = people.id ")
+      .select("reports.*, people.diaspora_handle as reported_author, people.guid as reported_author_guid")
+  }
 
   def reported_author
-    item.author if item
+    return Person.find(reported_author_id) if reported_author_id.present?
+    item&.author
   end
 
   def entry_does_not_exist
-    return unless Report.where(item_id: item_id, item_type: item_type).exists?(user_id: user_id)
-
-    errors.add(:base, "You cannot report the same post twice.")
+    if Report.where(item_id: item_id, item_type: item_type).exists?(user_id: user_id)
```

Comment:
Style/GuardClause: Use a guard clause (`return unless Report.where(item_id: item_id, item_type: item_type).exists?(user_id: user_id)`) instead of wrapping the code inside a conditional expression.